src/index.ts

```diff
@@ -87,11 +87,11 @@ const getAssetFromKV = async (event: FetchEvent, options?: Partial<Options>) =>
   const SUPPORTED_METHODS = ['GET', 'HEAD']
 
   if (!SUPPORTED_METHODS.includes(request.method)) {
-    throw new Error(`${request.method} is not a valid request method`)
+    throw new KVError(`${request.method} is not a valid request method`, 405)
```

Comment:
i think i commented with this elsewhere so maybe i am in the minority, but switching on http status codes rather than custom error types feels somehow wrong...

my expectation was to see discreet error types for each error that we might throw, e.g. `KVNotFoundError` and `KVInvalidMethodError`, etc.

Reply:
What's the advantage of multiple custom Error codes? Since we only throw 3 errors in the function in seems like overkill and it'd be nice to keep them all consistent type so the user knows how to generate a Response

Reply:
switching on a code vs a type feels like approximately the same thing but without the advantage of the type; and we should be documenting the possible codes you would get back and that's about the same cognitive overhead as separate errors.

Reply:
+1

Reply:
+1